compiler/modules/types/core.bal

```diff
@@ -705,21 +706,7 @@ function maybeRoDiff(SemType t1, SemType t2, Context? cx) returns SemType {
     BasicSubtype[] subtypes = [];
     foreach var [code, data1, data2] in new SubtypePairIteratorImpl(t1, t2, some) {
         SubtypeData data;
-        if cx == () || code < BT_COUNT_INHERENTLY_IMMUTABLE {
-            // normal diff or read-only basic type
-            if data1 == () {
-                var complement = ops[code].complement;
-                data = complement(<SubtypeData>data2);
-            }
-            else if data2 == () {
-                data = data1;
-            }
-            else {
-                var diff = ops[code].diff;
-                data = diff(data1, data2);
-            }
-        }
-        else {
+        if cx != () && (code == BT_LIST || code == BT_TABLE) {
```

Comment:
Needed to keep the `cx != ()` check as we call `isEmpty()` inside. Can get rid of the cx, once the list and table are integrated with cells.